### ISSUE TITLE
preprocessing: refactor mmpreprocess as a plugin

### ISSUE THREAD

**Issue by julou** [fjug/MoMA#19]
since mmpreprocess is built around imageJ, would it be easily possible to refactor it a plugin rather than a standalone executable? what I have in mind would be a plugin that would be given one image (or one frame of an hyperstack) and would create one roi per growth channel (ideally named filename_GLxx)? 
Then it would be straightforward to crop and save one dataset per GL…

In my opinion, it would be much more user friendly because it would keep all the preprocessing in Fiji. It might also help preserving image metadata during the preprocessing (#3). Other issues (e.g. #14) would be addressed directly by the user.

**Comment by julou** [comment]
in this case, the plugin get a more descriptive name, e.g. GrowthChannelFinder or GCfinder
because in practice the preprocessing involves anymore steps…

**Comment by fjug** [comment]
I agree, it would be nice to have this and I do not see technical hurdles. We would only need to find somebody that can afford spending the time doing it. 
We can discuss options, but potentially that could be done by one of our de.NBI people once we hired them... we'll have to see.

**Comment by julou** [comment]
are we talking hours / days / weeks of work? I mean, for someone knowing the IJ development environment, not someone like me!

**Comment by fjug** [comment]
I would say at least a week full time for someone who is totally into that kind of things. Two for someone who needs to understand what exists already. (Rough guesses, just to give you an idea...)
